gossipsub.go

```diff
@@ -56,32 +63,44 @@ func NewGossipSub(ctx context.Context, h host.Host, opts ...Option) (*PubSub, er
 // to use for injecting our messages in the overlay with stable routes; this
 // is the fanout map. Fanout peer lists are expired if we don't publish any
 // messages to their topic for GossipSubFanoutTTL.
-type GossipSubRouter struct {
+type GossipConfigurableRouter struct {
```

Comment:
Don't rename the struct/make a new one, just keep and extend the existing one.

Reply:
The new struct breaks the contract of the old struct, but reuses almost all of its code. I don't really care what the struct is named, but anyone who was using `GossipSubRouter` instead of `NewGossipSub` is going to be quite surprised.

Reply:
what? These are all internal in the library, I don't care if we break the old struct.

Reply:
as mentioned below I'm fine changing it back since we're fine breaking the struct contract.